pkg/adsc/adsc.go

```diff
@@ -474,32 +474,38 @@ func (a *ADSC) handleRecv() {
 		clusters := []*cluster.Cluster{}
 		routes := []*route.RouteConfiguration{}
 		eds := []*endpoint.ClusterLoadAssignment{}
-		for _, rsc := range msg.Resources { // Any
-			a.VersionInfo[rsc.TypeUrl] = msg.VersionInfo
-			valBytes := rsc.Value
-			switch rsc.TypeUrl {
-			case v3.ListenerType:
+		a.VersionInfo[msg.TypeUrl] = msg.VersionInfo
+		switch msg.TypeUrl {
```

Comment:
We can move https://github.com/istio/istio/pull/29296/files#diff-b5f677f06a05070210449a3f61f748b01824b494d5cf4116f88e50606c621d47R515-R523  here

Reply:
I didn't get it. Could you please point me to the codes?

Reply:
I mean the two switch case can be merged